libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/MapRouteLine.kt

```diff
@@ -1160,6 +1181,16 @@ internal class MapRouteLine(
                 it.addStringProperty(WAYPOINT_PROPERTY_KEY, propValue)
             }
         }
+
+        fun findDistanceOfPointAlongLine(line: LineString, point: Point): Double {
+            val pointOnLine = TurfMisc.nearestPointOnLine(point, line.coordinates()).geometry() as Point
```

Comment:
Good find on the `lineSlice` method below! Looking at the docs and impl, it already calls `nearestPointOnLine` for both arguments, so we can remove this line and rely only on the `lineSlice` to avoid duplicating the calculations.

Reply:
updated

Reply:
I'm seeing the execution take between 0 and 1 ms on average.